Tests/TestingTests/ExitTestTests.swift

```diff
@@ -0,0 +1,93 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+@testable @_spi(Experimental) @_spi(ForToolsIntegrationOnly) import Testing
+private import TestingInternals
+
+#if SWIFT_PM_SUPPORTS_SWIFT_TESTING && canImport(Foundation) && (os(macOS) || os(Linux) || os(Windows))
+@Suite("Exit test tests") struct ExitTestTests {
+  @Test("Exit tests (passing)") func passing() async {
+    await #expect(exitsWith: .failure) {
+      exit(EXIT_FAILURE)
+    }
+    if EXIT_SUCCESS != EXIT_FAILURE + 1 {
+      await #expect(exitsWith: .failure) {
+        exit(EXIT_FAILURE + 1)
+      }
+    }
+    await #expect(exitsWith: .success) {
+      exit(EXIT_SUCCESS)
+    }
+    await #expect(exitsWith: .exitCode(123)) {
+      exit(123)
+    }
+#if !os(Windows)
+    await #expect(exitsWith: .signal(SIGKILL)) {
+      _ = kill(getpid(), SIGKILL)
+    }
+    await #expect(exitsWith: .signal(SIGABRT)) {
+      abort()
+    }
+#endif
+  }
+
+  @TaskLocal
+  static var isTestingFailingExitTests = false
+
+  @Test("Exit tests (failing)") func failing() async {
+    let expectedCount: Int
+#if os(Windows)
+    expectedCount = 4
+#else
+    expectedCount = 6
+#endif
+    await confirmation("Exit tests failed", expectedCount: expectedCount) { failed in
+      var configuration = Configuration()
+      configuration.eventHandler = { event, _ in
+        if case .issueRecorded = event.kind {
+          failed()
+        }
+      }
+
+      await Self.$isTestingFailingExitTests.withValue(true) {
+        await Runner(selecting: "failingExitTests()", configuration: configuration).run()
+      }
+    }
+  }
+}
+
+// MARK: - Fixtures
+
+// This fixture can't be .hidden because it needs to be discovered correctly
```

Comment:
The child process is given an explicit identifier of the test to run, right? I thought even .hidden tests could be selected to run that way?

Reply:
`--filter` cannot select hidden tests by design because hidden tests are not used outside of swift-testing's own test target.